### ISSUE TITLE
GDB Debugger: Workspace IDE freeze when debugging noexit program

### ISSUE THREAD

**Issue by schervet** [eclipse-che/che#2508]
The workspace IDE get frozen when GDB Debugger is launched with noexit program (embedded programs, daemon, ...).

**Reproduction Steps:**
1. create a workspace from C++ stack with c-simple-console project.
2. add a while loop in Hello.c, e.g:

```
    int main() {
        printf("Hello World\n");
        while(1);
        return 0;
    }
```
1. Compile "cd ${current.project.path} && gcc -o a.out Hello.c"
2. In "Debug Configurations", select GDB and click on "Debug local binary"
3. click on "Debug"

**Expected behavior:**

When GDB Debugger is running (because you forgot to add breakpoint on noexit program), it should be possible to stop it (suspend, disconnect...).

**Observed behavior:**

When GDB Debugger is running, the "Loading..." animation is freezing the workspace (Not possible to click on debug toolbar or Main Menu). The only way to get back to workspace is to refresh the browser, quickly click on "End debug session" and then refresh the browser again.

**Che version:**    tested on 4.6 (Local Install Yatta), 4.7 (codenvy), 5.0.0-M1 (Local Install Yatta)
**OS and version:** VirtualBox ubuntu 14.04 64 bits
**Docker version:** 1.12.1
**Che install:**    Yatta   

**Additional information:**
- Problem started happening recently, didn't happen in an older version of Che: No
- Problem can be reliably reproduced, doesn't happen randomly: Yes

**Comment by TylerJewell** [comment]
@tolusha @eivantsov - this is an interesting use case. We just went through a variety of these scenarios with GDB + Artik, so surprised this particular issue is happening.

**Comment by ghost** [comment]
https://github.com/codenvy/artik-ide/issues/146

Yes, this is how gdb behaves. But agree, we need to be able to exit the debug session in such a case. Although the native gdb will hang as well.

The simplest solution is to remove loader and make it possible to end debug session with stop button on the debugger panel, and if this fails kill gdb process.

**Comment by schervet** [comment]
Thanks for your reactivity! As you said for a quick fix I removed loader and I also placed a 10s timeout on request (files attached).
[QuickFix CHE-2508.zip](https://github.com/eclipse/che/files/484443/QuickFix.CHE-2508.zip)

Just to give you the context, I am working on embedded devices like ARM Cortex-M and I started to use Eclipse Che for a Demo (I am in contact with  @slemeur on this topic). I use OpenOCD as gdb-server, and I have created a debugger plugin derivated from plugin-gdb to see if it could work in this specific use case (as you said the use case seems quite close from codeny/artik-ide, I will have a look at it). I am currently cleaning my work to make it available to the community in the next days.

I have also suggestions to create a "suspend" action, with possibility to call SIGINT signal in ProcessManager, is it better for you that I create new tickets for it or to track this suggestions here?
